pytorch_lightning/loops/fit_loop.py

```diff
@@ -205,6 +205,10 @@ def reset(self) -> None:
 
     def on_run_start(self) -> None:  # type: ignore[override]
         """Calls the ``on_train_start`` hook."""
+        # update the current_epoch in-case of checkpoint reload
+        if not self._iteration_based_training():
```

Comment:
during restart if it's not iteration-based training, we need to update the current epoch so that it starts from a fresh epoch rather than the old one.

Reply:
yes! good catch! will update

Reply:
> Does this make the following 2 lines unnecessary?

maybe... will check. I remember a test was failing: https://github.com/PyTorchLightning/pytorch-lightning/runs/6101153350?check_suite_focus=true

but it should not I guess since we increment `current.completed` after every epoch.

Reply:
okay looks like `done` is called within `skip`, so we need to keep it.

Reply:
Hey @rohitgr7 
I don't understand this logic. There is no true iteration based training in Lightning and we always have epochs. We may restart from a completed epoch or from an incomplete epoch, regardless of how the max_ flags on the trainer is set.

Reply:
if a user is using `max_steps=7` and uses a checkpoint from `step=4` to restart, we need to start from step 5 in that case. Although I guess the dataloaders are re-iterated from the beginning in that case.

Reply:
This explanation is not satisfying. It doesn't answer how max_steps has anything to do with the way we restart. 

You could also have `max_epochs=1` where the epoch size is 7 and you would still restore the checkpoint on step 4 the exact same way. 

max_steps / max_epochs are the stopping conditions. Them affecting the way we restart is beyond my understanding.

Reply:
> You could also have max_epochs=1 where the epoch size is 7 (equivalent to max_steps=7) and you would still restore the checkpoint on step 4 the exact same way.

yes, it does but in that case we do start from step=4 but on an entirely new epoch. The reason we need to separate this a little is that we don't need to update the current epoch. Although just noticed that we do update the current_epoch at the end even if the user is performing training based on `max_steps`. Do you think we should remove this condition and keep incrementing the current epoch on each restart even though the training is based on `max_steps`?

Reply:
Whatever the fix is, it will not at all be conditioned on max_steps / max_epochs trainer flags. I don't know how to solve this problem and we need to brainstorm this what should be fixed. 

All I know right now is that this PR did something weird 😅 that can't be the fix.